Add admin features (CRUD)

ttoino opened the pull request:
Closes #30.

The created pages can be found at /admin/<model>s. Example: /admin/users

ttoino committed:
Add users CRUD

Naapperas commented:
Is there anything missing in this PR @ttoino?

ttoino committed:
Add editions CRUD

ttoino committed:
Fix user edit page

ttoino committed:
Add events CRUD

ttoino committed:
Add speakers CRUD, fix users CRUD

ttoino committed:
Add products CRUD

ttoino committed:
Show product edition

ttoino committed:
Add quests CRUD

ttoino committed:
Fix type checking and format errors

ttoino commented:
> Is there anything missing in this PR @ttoino?

I think it's done.

Naapperas commented on resources/js/Components/CRUD/Row.vue:
```diff
@@ -0,0 +1,27 @@
+import route from "ziggy-js";
+
+defineProps<{
+    item: Model;
```
Why didn't you use a generic here like you did in the Table component?

Naapperas replied:
Also, another nitpick: in other places you create an interface called Props but here you are defining the type inline, not really important but it's a matter of consistency.

ttoino replied:
I didn't use a generic because that particular element does not need any properties outside the `Model` interface and it doesn't expose that prop anywhere else so being a generic wouldn't be advantageous.

Naapperas reviewed (dismissed):
LGTM. I believe that the pages should live under an `admin` folder and not a `CRUD` folder but that is not really that important right now.

Naapperas commented:
The tests are failing with the given message:
`Class "Illuminate\Support\Facades\App\Models\Student" not found`

Perhaps we messed up some paths, I believe this is not a problem with the logic itself of the website.

ttoino committed:
Fix user factories

ttoino commented:
> The tests are failing with the given message: `Class "Illuminate\Support\Facades\App\Models\Student" not found`
> 
> Perhaps we messed up some paths, I believe this is not a problem with the logic itself of the website.

It was a problem with the factories, I added the usertype to the $with property, but the id was 0, causing an exception. It's now fixed.

ttoino committed:
Fix props not an interface

ttoino committed:
Add factories to all models

ttoino committed:
Fix error in UserFactory

ttoino committed:
Add seed script

Naapperas commented on app/Http/Controllers/CRUDController.php:
Maybe we could add a `getById` method to CRUD Controllers? I can see it being useful.

Naapperas replied:
Besides that, you should write some documentation to explain what each thing does., for future reference.

ttoino replied:
> Maybe we could add a `getById` method to CRUD Controllers? I can see it being useful.

Sure, that'll be useful in the future if we need more in depth info for some models, like for validating quest requirements or forcing a user to renew their password.

> Besides that, you should write some documentation to explain what each thing does., for future reference.

Makes sense.

Naapperas commented on app/Http/Controllers/CRUDController.php:
```diff
@@ -0,0 +1,102 @@
+        $this->model::find($id)->delete();
+
+        return redirect()->back();
+    }
```
Shouldn't we also have a `destroyed` method?

ttoino replied:
I can't see how that would be useful, but I'll add it for completeness.

Naapperas replied:
It might not be needed at all, its just a matter of completeness as you said

Naapperas reviewed (changes requested):
LGTM :rocket:

ttoino committed:
Add show and destroyed methods, document CRUD

ttoino committed:
Make seed script executable, fix company factory

ttoino committed:
Format code
